File: frontend/js/controller.js
const controller = {
    proximoPasso: function () {
        fetch("http://localhost:8000/step", { method: "POST" })
            .then(response => response.json())
            .then(model => view.atualizarInterface(model));
    },

    resetar: function () {
        fetch("http://localhost:8000/reset", { method: "POST" })
            .then(response => response.json())
            .then(model => view.atualizarInterface(model));
    },

    carregarModelo: function () {
        fetch("http://localhost:8000/model")
            .then(response => response.json())
            .then(model => view.atualizarInterface(model));
    }
};

File: backend-java/src/Controller.java
public class Controller {
    private Model model;

    public Controller() {
        model = new Model();
    }

    public String obterModeloComoJson() {
        return model.toJson();
    }

    public String executarPasso() {
        model.executarPasso();
        return model.toJson();
    }

    public String reiniciar() {
        model.resetar();
        return model.toJson();
    }
}
